contracts/WildcardSteward_v2.sol

```diff
@@ -466,17 +530,110 @@ contract WildcardSteward_v2 is Initializable {
         address currentOwner = assetToken.ownerOf(tokenId);
         address tokenPatron = currentPatron[tokenId];
 
-        if (state[tokenId] == StewardState.Owned) {
-            uint256 totalToPayBack = price[tokenId];
-            // NOTE: pay back the deposit only if it is the only token the patron owns.
-            if (
-                totalPatronOwnedTokenCost[tokenPatron] ==
-                price[tokenId].mul(patronageNumerator[tokenId])
-            ) {
-                totalToPayBack = totalToPayBack.add(deposit[tokenPatron]);
-                deposit[tokenPatron] = 0;
-            }
+        _buy(tokenId, tokenPatron, wildcardsPercentage);
+        _handleBuy(
+            tokenId,
+            wildcardsPercentage,
+            remainingValueForDeposit,
+            currentOwner,
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    // Check possible issues with collectPatronage(tokenId) [probably leave out since no patronage to collect on forecolsed token!?]
+    function buyAuction(
+        uint256 tokenId,
+        uint256 _newPrice,
+        uint256 wildcardsPercentage
+    )
+        public
+        payable
+        collectPatronage(tokenId)
+        collectPatronageAddress(msg.sender)
+        priceGreaterThanZero(_newPrice)
```

Comment:
That is enforced by it being a `uint` - we could have a "not greater than 100000 Eth" rule so that someone can't set the price crazy high by eccident?

Reply:
Nevermind, I see that it is to litrally prevent the price being 0, eg 1wei is fine.